File: TalkBackGameServer/service/PlayerService.js
import Player from "/../models/Player.js";

class PlayerService {
  async addPlayer(user) {
    const addedPlayer = await Player.create({ ...user });
    return addedPlayer;
  }

  async getPlayers() {
    const players = await Player.find();
    return players;
  }

  async getPlayerByUsername(username) {
    if (!username) {
      throw new Error("Username is not specified");
    }

    const player = await Player.findOne({ username: username });
    return player;
  }

  async getPlayerById(playerId) {
    if (!playerId) {
      throw new Error("ID is not specified");
    }

    const player = await Player.findOne({ _id: playerId });
    return player;
  }

  async updatePlayer(player) {
    if (!player._id) {
      throw new Error("ID is not specified");
    }
    const updatedPlayer = await Player.findByIdAndUpdate(player._id, player, {
      new: true,
    });

    return updatedPlayer;
  }

  async deletePlayer(id) {
    if (!id) {
      throw new Error("ID is not specified");
    }

    const deletedPlayer = await Player.findByIdAndDelete(id);

    return deletedPlayer;
  }
}

export default new PlayerService();

File: server/game/modules/TicTacToe/Services/PlayerService.cs
using TicTacToe.Models;

namespace TicTacToe.Services
{
    public class PlayerService : IPlayerService
    {
        public Player[] GetPlayers(string player1Username, string player2Username)
        {
            var players = new Player[] { new Player(player1Username), new Player(player2Username) };
            return players;
        }

        public Player UpdatePlayer(Player player)
        {
            return player;
        }
    }
}
